pkg/xds/cache/mesh/snapshot.go

```diff
@@ -103,3 +105,22 @@ func hashResource(r core_model.Resource) string {
 				v.GetMeta().GetVersion()}, ":")
 	}
 }
+
+// We need to hash all the resolved IPs, not only the first one, because hostname can be resolved to two IPs (ex. LoadBalancer on AWS)
+// If we were to pick only the first one, DNS returns addresses in different order, so we could constantly get different hashes.
+func (m *meshSnapshot) hashResolvedIPs(address string) string {
+	if address == "" {
+		return ""
+	}
+	ips, err := m.ipFunc(address)
+	if err != nil {
+		// we can ignore an error and assume that address is not yet resolvable for some reason, once it will be resolvable the hash will change
```

Comment:
Do you think it makes sense to log?

Reply:
I'm on the fence here... but ok. Added log on V1. I don't think it's a good idea to log on V0. It's normal that hostname cannot be resolved for some reason yet. We will see an error in XDS generator anyway.